### PR TITLE
feat: Enhance session item UI

### LINKED ISSUE #5774: Schedule: Not needed comma if no position filled in by speaker
On the schedule a comma and space show up even though they are not required. This happens if the speaker has not filled in a position. Expected no comma and space should show up if the speaker has not filled in a position. Equally if the speaker has not filled in a company no comma and space should show up.

Compare https://eventyay.com/e/3dbaaa50/schedule

Screenshot from 2020-11-26 12-51-11

### LINKED ISSUE #5817: Public Schedule: Expanded session info - add position and organization, hide top profile icon and details on top when expanded
Some information is repeated in expanded sessions. The website generator hides some info on expanding sessions. Please implement something similar on the frontend. Compare http://2019.fossasia.org/event/schedule.html

Make changes as follows:
* When session is expanded hide profile image, name, designation and org on the top area
* Leave session info on top area as is
* On the expanded area show position and organization below the speaker name without brackets

Compare https://eventyay.com/e/8fa7fd14/schedule
Screenshot from 2020-11-29 00-56-08

### LINKED ISSUE #5818: Public Schedule: Show "Download Slides/Link to Slides" button instead of link on expanded session
The expanded session currently shows a link to slides if they have been added. Please change the feature as follows:
1. If the speaker uploaded slides show a button "Download Slides". Also include an icon in the download button.
2. If the speaker has only linked to the slides show a button "Link to Slides". Also include a link icon as part of the button.

Compare https://eventyay.com/e/8fa7fd14/schedule
Screenshot from 2020-11-29 01-05-47

### PR DESCRIPTION
**Pull request by daretobedifferent18**
Fixes #5817 
Fixes #5774 
Fixes #5818

#### Changes proposed in this pull request:

  - [x] When session is expanded hide profile image, name, designation and org on the top area
  - [x] Leave session info on top area as is
  - [x] On the expanded area show position and organization below the speaker name without brackets
  - [x] If the speaker uploaded slides show a button "Download Slides". Also include an icon in the download button.
  - [x] If the speaker has only linked to the slides show a button "Link to Slides". Also include a link icon as part of the button.
  - [x] On the schedule a comma and space show up even though they are not required. This happens if the speaker has not filled in a position. Expected no comma and space should show up if the speaker has not filled in a position. Equally if the speaker has not filled in a company no comma and space should show up.

#### ScreenShots
Screenshot from 2020-11-29 11-48-53
Screenshot from 2020-11-29 11-49-26
---
Screenshot from 2020-11-29 15-05-39
---
Screenshot from 2020-11-29 17-41-37

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by daretobedifferent18** [comment]
@iamareebjamal It's Done.

**Review by sachinchauhan2889 — CHANGES_REQUESTED** [review]
microlocation and date-time information shift to left side after expanding session card. They must remain on right side even on expanding session.

**Comment by daretobedifferent18** [comment]
> microlocation and date-time information shift to left side after expanding session card. They must remain on right side even on expanding session.

That was actually a bug before which was not working but now its working correctly

**Comment by iamareebjamal** [comment]
#5774 is not fixed

**Comment by daretobedifferent18** [comment]
> #5774 is not fixed

Working on both #5774 #5818 
See the Checklist in the description

**Comment by daretobedifferent18** [comment]
@iamareebjamal only left with #5818 but in this issue actually the link is been added in short Abstract and not in Slide url, so it will remain as it is, I think we can't change that.
Pls let me know what to do

**Comment by daretobedifferent18** [comment]
@iamareebjamal Done.

**Review by mariobehling — DISMISSED** [review]
Nice work! Thanks

File: app/components/public/session-item.hbs [hunk 1]
```diff
@@ -40,7 +40,7 @@
            </div>
            <div class="left floated nine wide column">
              {{#each @session.speakers as |speaker|}}
-                {{speaker.name}} ({{speaker.position}}, {{speaker.organisation}})
+                {{speaker.name}} {{#if (or speaker.position speaker.positionOrganisation)}}({{speaker.positionOrganisation}}){{/if}}
```
Comment (iamareebjamal) [inline app/components/public/session-item.hbs:49]: Simply use `if speaker.positionOrganisation`

File: app/components/public/session-item.hbs [hunk 2]
```diff
@@ -5,6 +5,12 @@
       <h3 class="ui header" id="session-id-{{@session.id}}" style={{css color=(text-color @session.track.color)}}>
         {{@session.title}}
         <span style={{css float='right'}}>
+          {{#if @session.slidesUrl}}
+            <button class="ui basic {{text-color @session.track.color 'basic' 'inverted'}} button" style={{css color=(text-color @session.track.color 'grey' 'lightgrey')}} {{action this.goToSlides}}>
+              <i class="icon {{if this.slidesUploaded 'download' 'linkify'}}"></i>
+              {{t (if this.slidesUploaded 'Download Slides' 'Link to Slides')}}
```
Comment (iamareebjamal) [inline app/components/public/session-item.hbs:11]: t should have static string as argument. Please move if out and t in the conditions

**Comment by mariobehling** [comment]
Oh, in the last iteration the date and time now appear twice.

Compare https://open-event-frontend-git-session-expanded-5817.eventyay.vercel.app/e/8fa7fd14/schedule

Screenshot from 2020-11-29 14-50-54

File: app/components/public/session-item.hbs [hunk 3]
```diff
@@ -61,20 +67,9 @@
         {{sanitize @session.shortAbstract}}
       </div>
     </div>
-    {{#if (and @expanded (or @session.slidesUrl @session.videoUrl))}}
+    {{#if (and @expanded @session.videoUrl)}}
       <div class="row p-4">
-        <div class="column" style={{css display='flex' align-items='center' flex-direction='column'}}>
-          {{#if (or this.pdfLink this.pptLink)}}
-            <iframe title="session-slides" class="slide-iframe mb-2" width="100%" frameborder="0" src="https://docs.google.com/gview?url={{ @session.slidesUrl }}&embedded=true"></iframe>
```
Comment (iamareebjamal) [inline app/components/public/session-item.hbs:68]: We want to show slides here. Please revert

**Comment by daretobedifferent18** [comment]
@iamareebjamal Made Changes

File: app/components/public/session-item.hbs [hunk 4]
```diff
@@ -66,13 +72,6 @@
         <div class="column" style={{css display='flex' align-items='center' flex-direction='column'}}>
           {{#if (or this.pdfLink this.pptLink)}}
             <iframe title="session-slides" class="slide-iframe mb-2" width="100%" frameborder="0" src="https://docs.google.com/gview?url={{ @session.slidesUrl }}&embedded=true"></iframe>
-          {{else if @session.slidesUrl}}
-            <a class="mb-2" href={{ @session.slidesUrl }} target="_blank" rel="noopener noreferrer">
-              <button class="ui labeled icon button">
-                <i class="file powerpoint icon"></i>
-                {{t 'Slides'}}
-              </button>
-            </a>
```
Comment (iamareebjamal) [inline app/components/public/session-item.hbs:75]: Don't remove this from here please

Comment (daretobedifferent18) [reply app/components/public/session-item.hbs]: Done

**Comment by daretobedifferent18** [comment]
@iamareebjamal Pls review. This PR is completed

**Comment by iamareebjamal** [comment]
@mariobehling I am merging this, but session collapsed will look very cluttered when there is both video stream and slides attached to it

**Comment by mariobehling** [comment]
Yes, Thank you.

Let's talk about this in the weekly meeting.

**Pull request merged**